modules/imgproc/src/thresh.cpp

```diff
@@ -269,6 +269,204 @@ thresh_8u( const Mat& _src, Mat& _dst, uchar thresh, uchar maxval, int type )
     }
 }
 
+static void
+thresh_16u(const Mat& _src, Mat& _dst, ushort thresh, ushort maxval, int type)
+{
+	Size roi = _src.size();
+	roi.width *= _src.channels();
+	size_t src_step = _src.step;
+	size_t dst_step = _dst.step;
+
+	if (_src.isContinuous() && _dst.isContinuous())
+	{
+		roi.width *= roi.height;
+		roi.height = 1;
+		src_step = dst_step = roi.width;
+	}
+
+	// HAVE_TEGRA_OPTIMIZATION not supported
+
+	// HAVE_IPP not supported
+
+	int j = 0;
+	const ushort* src = _src.ptr<ushort>();
+	ushort* dst = _dst.ptr<ushort>();
+
+	// CV_SIMD128 not supported
+#if CV_SIMD128
+	bool useSIMD = checkHardwareSupport(CV_CPU_SSE2) || checkHardwareSupport(CV_CPU_NEON);
```

Comment:
There's a [hasSIMD128()]( https://github.com/opencv/opencv/blob/32d4af36e273dad3d603ba983669d76da946845f/modules/core/include/opencv2/core/hal/intrin_sse.hpp#L1808 ) which checks the feature in better way